src/connection_string.ts

```diff
@@ -228,7 +228,7 @@ function toRecord(value: string): Record<string, any> {
   return record;
 }
 
-class CaseInsensitiveMap extends Map<string, any> {
+class CaseInsensitiveMap<Value = any> extends Map<string, Value> {
```

Comment:
I added a type parameter here because I wanted stronger typing on the `urlOptions` object.  Since the urlOptions object is always `Map<string, any[]>`, I wanted Typescript to help out so that it's clear to future developers on line 332 why we can immediately spread the object (unlike the other two if cases).